tools/nd_freeze_install

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/python3
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Script to install older packages based on version number.
+#
+# If running this in a stripped down environment such as a docker container,
+# install python dependencies with this command:
+#
+# apt-get update && apt-get install -y python3 python3-pip && pip3 install requests bs4
+#
+import argparse
+import glob
+import gzip
+import logging
+import os
+import re
+import requests
+import subprocess
+from bs4 import BeautifulSoup
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "packages",
+    help="package and version to install (e.g. git=2.17.1)",
+    nargs="+",
+    metavar="package=version"
+)
+parser.add_argument(
+    "-n", "--no-updates",
+    help=("Make the snapshot repositories available but skip "
+        "the package update step, leaving the system as is."),
+    action="store_true",
+    dest="no_updates"
+)
+parser.add_argument(
+    "-d", "--debug",
+    help="print debugging informaiton to the screen",
+    action="store_true",
+    dest="debug"
+)
+parser.add_argument(
+    "-t", "--trust-repos",
+    help=("Update untrusted repos. For repos whose apt keys are invalid, "
+        "go ahead and use the repo rather thanfailing by default. "
+        "USE CAUTIOUSLY!"),
+    action="store_true",
+    dest="trust_repos"
+)
+parser.add_argument(
+    "-a", "--architecture",
+    help=("Choose the architecture of the package to install. "
+        "Defaults to host architecture."),
+    dest="architecture"
+)
+args = parser.parse_args()
+
+if (args.debug):
+    logging.basicConfig(level=logging.DEBUG)
+else:
+    logging.basicConfig(level=logging.INFO)
+
+
+# Run a shell command
+#
+# Parameters
+# ----------
+# command
+#   string : command to run in shell
+# env
+#   string : hash of environment variables to pass to command environment
+#
+# Returns
+# -------
+# string : STDOUT
+#
+def exec_shell(command, env=None):
+    logging.debug("Executing shell command: '{}'".format(command))
+    if env:
+        command_env = os.environ.copy()
+        command_env.update(env)
+    s = subprocess.Popen(command.split(), stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT, env=env)
+    stdout, stderr = s.communicate()
+    if stdout:
+        stdout = stdout.decode('utf-8')
+    if stderr:
+        stderr = stderr.decode('utf-8')
+    logging.debug("Command STDOUT: {}".format(stdout))
+    logging.debug("Command STDERR: {}".format(stderr))
```

Comment:
Should avoid spending time on creating formatted string if it wouldn't even be shown. Here and elsewhere:
```suggestion
    logging.debug("Command STDERR: %s", stderr)
```